test/Collector.test.ts

```diff
@@ -0,0 +1,526 @@
+        PARTNER_SHARES
+      );
+      const expectedTokens = fakeERC20Tokens.map(({ address }) => address)[0];
+      const expectedOwnerAddress = owner.address;
+
+      const collector: MockContract<Collector> = await collectorFactory.deploy(
+        expectedOwnerAddress,
+        [expectedTokens],
+        partners,
+        remainderDestination.address
+      );
+
+      const actualOwnerAddress = await collector.owner();
+      const actualTokens = await collector.getTokens();
+
+      expect(actualOwnerAddress, 'Failed to set the owner').to.equal(
+        expectedOwnerAddress
+      );
+      expect(actualTokens.toString(), 'Failed to set tokens').to.equal(
+        expectedTokens.toString()
+      );
+    });
+
+    it('Should deploy a Collector with multiple tokens', async function () {
+      partners = await buildPartners(
+        [partner1, partner2, partner3, partner4],
+        PARTNER_SHARES
+      );
+      const expectedTokens = fakeERC20Tokens.map(({ address }) => address);
+      const expectedOwnerAddress = owner.address;
+
+      const collector: MockContract<Collector> = await collectorFactory.deploy(
+        owner.address,
+        expectedTokens,
+        partners,
+        remainderDestination.address
+      );
+      const actualOwnerAddress = await collector.owner();
+      const actualTokens = await collector.getTokens();
+
+      expect(actualOwnerAddress, 'Failed to set the owner').to.equal(
+        expectedOwnerAddress
+      );
+      expect(actualTokens.toString(), 'Failed to set tokens').to.equal(
+        expectedTokens.toString()
+      );
+    });
+
+    it('Should not deploy with invalid shares', async function () {
+      const partners = await buildPartners(
+        [partner1, partner2, partner3, partner4],
+        [25, 25, 25, 26]
+      );
+
+      await expect(
+        collectorFactory.deploy(
+          owner.address,
+          fakeERC20Tokens.map(({ address }) => address),
+          partners,
+          remainderDestination.address
+        )
+      ).to.be.rejectedWith('Shares must add up to 100%');
+    });
+
+    it('Should not deploy if the array of partners is empty', async function () {
+      const partners: Partner[] = [];
+
+      await expect(
+        collectorFactory.deploy(
+          owner.address,
+          fakeERC20Tokens.map(({ address }) => address),
+          partners,
+          remainderDestination.address
+        )
+      ).to.be.rejectedWith('Shares must add up to 100%');
+    });
+
+    it('Should not deploy if a share is 0', async function () {
+      const partners = await buildPartners(
+        [partner1, partner2, partner3, partner4],
+        [30, 30, 0, 40]
+      );
+
+      await expect(
+        collectorFactory.deploy(
+          owner.address,
+          fakeERC20Tokens.map(({ address }) => address),
+          partners,
+          remainderDestination.address
+        )
+      ).to.be.rejectedWith('0 is not a valid share');
+    });
```

Comment:
Shouldn't we add a test case to check the deployment if no tokens are specified?

Reply:
As discussed, no.